### PR TITLE
Binary Search

### PR DESCRIPTION
**Pull request by AShaw110**
_**Implemented iterative binary search algorithm**_ in all the three languages. I do not know python, so had to use the code converter. Also wrote a decent explanation of how time complexity is calculated for it in the readme file.

### CODE REVIEW

File: Array_Functions/Binary_Search/binary_search.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
+//Implementatiom of Binary Search Algorithm in C++
+#include <bits/stdc++.h>
+using namespace std;
+
+int bsearch(double arr[], int n, double x)
+{
+    int mid, low = 0, high = n - 1;
+    while (low <= high)
+    {
+        mid = (low + high) / 2;
```
Comment (imsuraj675) [inline Array_Functions/Binary_Search/binary_search.cpp]: Good, your solution is correct. Only a slight modification is required, to handle cases when 'n' is significantly large. In that case if you calculate mid = (low + high)/2, then the value may overflow. Instead, use mid = (high - low)/2 + low. This is the most standard practice which is used.

Modify this in all three files

Comment (AShaw110) [reply Array_Functions/Binary_Search/binary_search.cpp]: Yeah I saw that somewhere and it did not make sense to me at that time. Thanks, will make the necessary changes.

**Comment by AShaw110** [comment]
@imsuraj675 Done bhaiya. Do I have to update the pull request?

**Comment by imsuraj675** [comment]
Good work

**Pull request merged**